wagtail/wagtailforms/models.py

```diff
@@ -260,20 +260,28 @@ def process_form_submission(self, form):
             page=self,
         )
 
+    def render_landing_page(self, request, *args, **kwargs):
+        """
+        Renders the landing page.
+
+        You can override this method to return a different HttpResponse as
+        landing page. E.g. you could return a redirect to a separate page.
+        """
+        # TODO: It is much better to redirect to it
+        return render(
+            request,
+            self.get_landing_page_template(request),
+            self.get_context(request)
+        )
+
     def serve(self, request, *args, **kwargs):
         if request.method == 'POST':
             form = self.get_form(request.POST, request.FILES, page=self, user=request.user)
 
             if form.is_valid():
                 self.process_form_submission(form)
```

Comment:
Duplicate of comment on #2834
----

We need a way to ensure that the form_submission instance can be passed to the context of the landing page.

As per this outstanding item in #2833
> The FormSubmission instance should then be passed to the render context for the "landing page".

 the `render_landing_page` method should also accept a parameter `form_submission`.

Which should be generated by updating the line:
`self.process_form_submission(form)` 
to
`form_submission = self.process_form_submission(form)` 

Then 
`return self.render_landing_page(request, *args, **kwargs)`
to
`return self.render_landing_page(request, form_submission=form_submission, *args, **kwargs)`

Finally, the context should be updated in the render_landing_page return render eg.

```
 # TODO: It is much better to redirect to it
context = self.get_context(request)
context['form_submission'] = form_submission
    return render(
        request,
        self.get_landing_page_template(request),
         context
        )
```